### DISCUSSION TITLE
Cutting a mesh with surface

### DISCUSSION THREAD

**Discussion by kuriankattukaren** [pyvista/pyvista#3363] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/7378439/192134725-f14b2281-d4dd-4e69-afbe-79a56d2d5905.png)

The top surface is the 4D response and the bottom is the surface on which I would like to drape the 4D response. However, as can be seen in the pic the bottom mesh is larger than the 4D surface.

Is there an easy way for me to trim off the extra cells of the bottom mesh using the top mesh?

Cheers,
Kurian Kattukaren

**Comment by banesullivan** [comment, marked as answer]
There are a handful of ways to go about this. One way is: 

You could extrude the top surface to create a shell, then use that shell with the `clip_surface` method.

For example,

```py
import pyvista as pv
from pyvista import examples

top = examples.load_random_hills()
bottom = pv.Plane(
    center=(*top.center[:2], 0), 
    i_size=30, j_size=30, 
    i_resolution=100, j_resolution=100
)


extrusion = top.extrude((0, 0, -1), capping=False, )
clipped = bottom.clip_surface(extrusion, invert=False)

p = pv.Plotter(notebook=0, shape=(1,3))
p.add_text('Initial')
p.add_mesh(top)
p.add_mesh(bottom, style='wireframe', color='grey')

p.subplot(0, 1)
p.add_text('Clip with extruded bounds')
p.add_mesh(extrusion, opacity=0.75)
p.add_mesh(bottom, style='wireframe', color='grey')

p.subplot(0, 2)
p.add_text('Result')
p.add_mesh(top)
p.add_mesh(clipped, style='wireframe', color='grey')

p.link_views()
p.view_isometric()
p.show()
```

<img width="1445" alt="Screen Shot 2022-09-26 at 11 34 13 AM" src="https://user-images.githubusercontent.com/22067021/192343055-8f741731-0b4d-4123-beb1-1b77e913811a.png">


Ref:

- https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.extrude.html?highlight=extrude#pyvista.PolyDataFilters.extrude
- https://docs.pyvista.org/examples/01-filter/clipping-with-surface.html

**Reply by kuriankattukaren** [reply]
@banesullivan

It worked like a charm.

![image](https://user-images.githubusercontent.com/7378439/192531597-495e4daa-368a-4ecc-85d5-7d50081ad51c.png)


Thanks for the answer. Is it worth making a pull request for the Pyvista examples category?

Cheers,
Kurian

**Reply by banesullivan** [reply]
Glad this worked out for you!

> Is it worth making a pull request for the Pyvista examples category?

Additions to the examples are always welcome!

I'm not sure whether it makes sense to add this on to an existing example like the one for `clip_surface`: https://docs.pyvista.org/examples/01-filter/clipping-with-surface.html

I'll let you decide!

**Comment by kuriankattukaren** [comment]
@banesullivan 

It worked like a charm.

![image](https://user-images.githubusercontent.com/7378439/192531006-25baa945-a692-41ba-8b73-628e46ea810c.png)

Thanks for the answer.  Is it worth making a pull request for the Pyvista examples category?

Cheers,
Kurian
